os/net/local/local_recvutils.c

```diff
@@ -0,0 +1,249 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <tinyara/fs/fs.h>
+
+#include "local/local.h"
+
+#if defined(CONFIG_NET) && defined(CONFIG_NET_LOCAL)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: local_fifo_read
+ *
+ * Description:
+ *   Read a data from the read-only FIFO.
+ *
+ * Input Parameters:
+ *   filep - File structure of write-only FIFO.
+ *   buf   - Local to store the received data
+ *   len   - Length of data to receive [in]
+ *           Length of data actually received [out]
+ *
+ * Returned Value:
+ *   Zero is returned on success; a negated errno value is returned on any
+ *   failure.  If -ECONNRESET is received, then the sending side has closed
+ *   the FIFO. In this case, the returned data may still be valid (if the
+ *   returned len > 0).
+ *
+ ****************************************************************************/
+
+int local_fifo_read(FAR struct file *filep, FAR uint8_t *buf, size_t *len)
+{
+	ssize_t remaining;
+	ssize_t nread;
+	int ret;
+
+	DEBUGASSERT(buf && len);
+
+	remaining = *len;
+	while (remaining > 0) {
+		nread = file_read(filep, buf, remaining);
+		if (nread < 0) {
+			if (nread != -EINTR) {
```

Comment:
file_read() returns -1 on failure with errno set appropriately.
```suggestion
			if (errno != EINTR) {
```

Reply:
I'll modify file_read